### PR TITLE
Use dotnet pack to generate NuGet package

### LINKED ISSUE #1146: Use dotnet pack to generate NuGet package
This will make the nuspec completely unnecessary (everything will be in the csproj). The dependencies are generated automatically, so we can no longer have dependency version inconsistencies between csproj and nuspec (see #1126).

Better solution to #1133

### PR DESCRIPTION
**Pull request by thomaslevesque**
Fixes #1146

### CODE REVIEW

File: build.csx [hunk 1]
```diff
@@ -66,7 +60,7 @@ static var xunit = "./packages/xunit.runner.console.2.0.0/tools/xunit.console.ex
 var coverityDirectory = "./artifacts/coverity";
 var coverityResultsDirectory = "./artifacts/coverity/cov-int";
 var logsDirectory = "./artifacts/logs";
-var outputDirectory = "./artifacts/output";
+var outputDirectory = Path.GetFullPath("./artifacts/output");
```
Comment (thomaslevesque) [inline build.csx]: Because of https://github.com/dotnet/cli/issues/4765

File: src/FakeItEasy/FakeItEasy.csproj [hunk 2]
```diff
@@ -10,10 +10,27 @@
...
+
   <ItemGroup>
     <CodeAnalysisDictionary Include="..\..\FakeItEasy.Dictionary.xml" />
     <Compile Include="../*.cs" />
-    <PackageReference Include="StyleCop.Analyzers" Version="1.0.0" />
+    <PackageReference Include="StyleCop.Analyzers" Version="1.0.0" PrivateAssets="all" />
+    <Content Include="$(OutputPath)/*/FakeItEasy.pdb">
+      <Pack>true</Pack>
+      <PackagePath>lib/%(RecursiveDir)FakeItEasy.pdb</PackagePath>
+    </Content>
```
Comment (thomaslevesque) [inline src/FakeItEasy/FakeItEasy.csproj]: This is a "hack" to include the pdbs in the same package

File: build.csx [hunk 3]
```diff
@@ -293,9 +293,10 @@ public void GenerateNuspec(string projectPath, string nuspecTemplatePath, string
         var references =
             from itemGroup in project.Root.Elements("ItemGroup")
             let condition = itemGroup.Attribute("Condition")?.Value.Trim()
-            where condition == $"'$(TargetFramework)' == '{tfm}'"
+            where condition == null || condition == $"'$(TargetFramework)' == '{tfm}'"
             from reference in itemGroup.Elements("PackageReference")
             where reference.Element("PrivateAssets") == null
+            && reference.Attribute("PrivateAssets") == null
```
Comment (blairconrad) [inline build.csx]: super nit: should this line be indented a little?

File: build.csx (hunk 1 above)
Comment (blairconrad) [inline build.csx]: *sadnod*

File: build.csx [hunk 4]
```diff
@@ -273,52 +261,6 @@ public string GetVSLocation()
...
-                    "dependency",
-                    new XAttribute("id", r.Id),
-                    new XAttribute("version", r.Version)));
-
-        dependenciesElement.Add(groupElement);
-    }
-
-    nuspec.Save(nuspecOutputPath);
-}
-
```
Comment (blairconrad) [inline build.csx]: Good riddance!

File: src/FakeItEasy/FakeItEasy.csproj [hunk 5]
```diff
@@ -10,10 +10,27 @@
     <CodeAnalysisRuleSet>..\..\FakeItEasy.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
 
+  <PropertyGroup Label="Package properties">
+    <Title>FakeItEasy - Mocking made faking easy!</Title>
+    <Authors>Patrik Hägne, FakeItEasy contributors</Authors>
+    <Description>It's faking amazing! The easy mocking framework for .NET that works great in C# and VB.NET alike. No need to know the difference between a stub, a mock or a spy, everything's a fake! The easy to use, refactoring friendly API makes faking a breeze.</Description>
+    <Copyright>Copyright (c) FakeItEasy contributors. (fakeiteasyfx@gmail.com)</Copyright>
+    <PackageProjectUrl>http://fakeiteasy.github.io/</PackageProjectUrl>
+    <PackageIconUrl>http://fakeiteasy.github.io/img/fakeiteasy_logo_256_square_white.png</PackageIconUrl>
```
Comment (blairconrad) [inline src/FakeItEasy/FakeItEasy.csproj]: outside the scope of the issue, but while you're moving the URLs, we could make them HTTPS…

I will not hold up the mege for this

File: src/FakeItEasy/FakeItEasy.csproj [hunk 6]
```diff
@@ -10,10 +10,27 @@
     <CodeAnalysisRuleSet>..\..\FakeItEasy.ruleset</CodeAnalysisRuleSet>
   </PropertyGroup>
 
+  <PropertyGroup Label="Package properties">
+    <Title>FakeItEasy - Mocking made faking easy!</Title>
+    <Authors>Patrik Hägne, FakeItEasy contributors</Authors>
+    <Description>It's faking amazing! The easy mocking framework for .NET that works great in C# and VB.NET alike. No need to know the difference between a stub, a mock or a spy, everything's a fake! The easy to use, refactoring friendly API makes faking a breeze.</Description>
+    <Copyright>Copyright (c) FakeItEasy contributors. (fakeiteasyfx@gmail.com)</Copyright>
+    <PackageProjectUrl>http://fakeiteasy.github.io/</PackageProjectUrl>
```
Comment (blairconrad) [inline src/FakeItEasy/FakeItEasy.csproj]: Is this really a less-than semicolon? `<;`

Comment (thomaslevesque) [inline src/FakeItEasy/FakeItEasy.csproj]: What do you mean? I don't get it...

File: src/FakeItEasy/FakeItEasy.csproj (hunk 5 above)
Comment (thomaslevesque) [inline src/FakeItEasy/FakeItEasy.csproj]: Well spotted. I'll change them.

File: build.csx (hunk 3 above)
Comment (thomaslevesque) [inline build.csx]: Well, this code is no longer there, so no :wink:

Comment (blairconrad) [inline build.csx]: HA! That's what I get for reviewing change at time. Tell me you at least indented it and then removed it? :blush:

File: src/FakeItEasy/FakeItEasy.csproj (hunk 6 above)
Comment (blairconrad) [inline src/FakeItEasy/FakeItEasy.csproj]: image

Comment (blairconrad) [inline src/FakeItEasy/FakeItEasy.csproj]: I checked the raw file. It's not a semicolon.

Comment (thomaslevesque) [inline src/FakeItEasy/FakeItEasy.csproj]: Where did you see that?

**Comment by thomaslevesque** [comment]
Squashed.

**Comment by blairconrad** [comment]
Stellar job. Thanks!

File: src/FakeItEasy/FakeItEasy.csproj (hunk 6 above)
Comment (blairconrad) [inline src/FakeItEasy/FakeItEasy.csproj]: Right in the csproj. Line 18. On the conversation _and_ files changed tab!

Comment (blairconrad) [inline src/FakeItEasy/FakeItEasy.csproj]: https://github.com/thomaslevesque/FakeItEasy/blob/19f6cbcd444c443949eb3a9f43ce52d4eebb65e4/src/FakeItEasy/FakeItEasy.csproj#L19

Comment (thomaslevesque) [inline src/FakeItEasy/FakeItEasy.csproj]: Weird. I don't see that.

image

**Comment by thomaslevesque** [comment]
Thanks for the merge!

File: src/FakeItEasy/FakeItEasy.csproj (hunk 6 above)
Comment (blairconrad) [inline src/FakeItEasy/FakeItEasy.csproj]: https://chrome.google.com/webstore/detail/refined-github/hlepfoohegkhhmjieoechaddaejaokhf?utm_source=chrome-app-launcher-info-dialog is the culprit

Comment (thomaslevesque) [inline src/FakeItEasy/FakeItEasy.csproj]: That's a weird bug...

Comment (blairconrad) [inline src/FakeItEasy/FakeItEasy.csproj]: https://github.com/sindresorhus/refined-github/issues/513.

**Pull request merged**